### ISSUE TITLE
Dataset dir of Val is same with Test, but the score not equal.

### ISSUE THREAD

**Issue by YancyGuo** [open-mmlab/mmaction2#2261]
Before raising a question, **I have checked** the following listed items.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmaction2.readthedocs.io/en/latest/faq.html) but cannot get the expected help.

**Q**
Before I used my dataset on models of mmaction2, I have tested the UCF101 and checked it work--the **Val score TOP1_acc** which in the Training, is equal to **Test value** because of the same dataset dir in config.
But when I modified the config to fit my binary classification dataset, the value of two score will amazing unequal and huge different--such as 
**Trainning val**
2023-03-03 10:27:10,020 - mmaction - INFO - Epoch [30][60/61]   lr: 3.750e-04, eta: 0:07:57, time: 0.351, data_time: 0.001, memory: 10391, **top1_acc: 0.9417**, top5_acc: 1.0000, loss_cls: 0.1357, loss: 0.1357, grad_norm: 4.4950
**Test** 
**top1_acc        0.6987**

It will show although over fitting performed on Val but not same acc on Test

I output my config here, and if you need more info and I will response as quite as possible ,thank U!!! 
BTW, although the output model is based on kinetics400，but I have tested both dataset former and ucf101 even 10 models in mmaction2. Which is the same result for this problem.

```php
model = dict(
    type='Recognizer2D',
    backbone=dict(
        type='ResNetTSM',
        pretrained='torchvision://resnet50',
        depth=50,
        norm_eval=False,
        shift_div=8,
        num_segments=16),
    cls_head=dict(
        type='TSMHead',
        num_classes=2,
        in_channels=2048,
        spatial_type='avg',
        consensus=dict(type='AvgConsensus', dim=1),
        dropout_ratio=0.5,
        init_std=0.001,
        is_shift=True,
        num_segments=16),
    train_cfg=None,
    test_cfg=dict(average_clips='prob'))
optimizer = dict(
    type='SGD',
    constructor='TSMOptimizerConstructor',
    paramwise_cfg=dict(fc_lr5=True),
    lr=0.00375,
    momentum=0.9,
    weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=20, norm_type=2))
lr_config = dict(policy='step', step=[20, 40])
total_epochs = 50
checkpoint_config = dict(interval=5)
log_config = dict(interval=20, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'checkpoints/tsm_r50_256p_1x1x16_50e_kinetics400_rgb_20201010-85645c2a.pth'
resume_from = None
workflow = [('train', 1)]
opencv_num_threads = 0
mp_start_method = 'fork'
dataset_type = 'RawframeDataset'
data_root = 'data/US_test/frames/train'
data_root_val = 'data/US_test/frames/val'
ann_file_train = 'data/US_test/frames/k400_train_rawframes.txt'
ann_file_val = 'data/US_test/frames/k400_val_rawframes.txt'
ann_file_test = 'data/US_test/frames/k400_val_rawframes.txt'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_bgr=False)
train_pipeline = [
    dict(type='SampleFrames', clip_len=1, frame_interval=1, num_clips=16),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(
        type='MultiScaleCrop',
        input_size=224,
        scales=(1, 0.875, 0.75, 0.66),
        random_crop=False,
        max_wh_scale_gap=1,
        num_fixed_crops=13),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(type='Flip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=16,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=16,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='TenCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
data = dict(
    videos_per_gpu=6,
    workers_per_gpu=2,
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type='RawframeDataset',
        ann_file='data/US_test/frames/k400_train_rawframes.txt',
        data_prefix='data/US_test/frames/train',
        pipeline=[
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=16),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(
                type='MultiScaleCrop',
                input_size=224,
                scales=(1, 0.875, 0.75, 0.66),
                random_crop=False,
                max_wh_scale_gap=1,
                num_fixed_crops=13),
            dict(type='Resize', scale=(224, 224), keep_ratio=False),
            dict(type='Flip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs', 'label'])
        ]),
    val=dict(
        type='RawframeDataset',
        ann_file='data/US_test/frames/k400_val_rawframes.txt',
        data_prefix='data/US_test/frames/val',
        pipeline=[
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=16,
                test_mode=True),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs'])
        ]),
    test=dict(
        type='RawframeDataset',
        ann_file='data/US_test/frames/k400_val_rawframes.txt',
        data_prefix='data/US_test/frames/val',
        pipeline=[
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=16,
                test_mode=True),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(type='TenCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs'])
        ]))
evaluation = dict(
    interval=5, metrics=['top_k_accuracy', 'mean_class_accuracy'])
work_dir = './work_dirs/tsm_r50_1x1x16_50e_kinetics400_rgb/'
gpu_ids = range(0, 2)
omnisource = False
module_hooks = []
```

**Comment by hukkai** [comment]
@YancyGuo The val performance should be lower than test performance as the test pipeline is using 10 crops where val pipeline is using 1 crop. I guess the accuracy shown here is `training accuracy`:

> 2023-03-03 10:27:10,020 - mmaction - INFO - Epoch [30][60/61] lr: 3.750e-04, eta: 0:07:57, time: 0.351, data_time: 0.001, memory: 10391, top1_acc: 0.9417, top5_acc: 1.0000, loss_cls: 0.1357, loss: 0.1357, grad_norm: 4.4950

If it is the validation accuracy, the `grad_norm` should not be displayed.

**Comment by hukkai** [comment]
@YancyGuo The validation accuracy is displayed as:
>  Epoch(val) [15][3757]	top1_acc: 0.7692, top5_acc: 0.9291, mean_class_accuracy: 0.7690

The the answer is that the model you trained is overfitting.

**Comment by YancyGuo** [comment]
> @YancyGuo The val performance should be lower than test performance as the test pipeline is using 10 crops where val pipeline is using 1 crop. I guess the accuracy shown here is `training accuracy`:
> 
> > 2023-03-03 10:27:10,020 - mmaction - INFO - Epoch [30][60/61] lr: 3.750e-04, eta: 0:07:57, time: 0.351, data_time: 0.001, memory: 10391, top1_acc: 0.9417, top5_acc: 1.0000, loss_cls: 0.1357, loss: 0.1357, grad_norm: 4.4950
> 
> If it is the validation accuracy, the `grad_norm` should not be displayed.

Yes I see, It's different from mmclassification project which will show val automatically, I have add --validate and it worked~ Thanks a lot!!
